osquery/tables/system/windows/scheduled_tasks.cpp

```diff
@@ -112,8 +113,19 @@ void enumerateTasksForFolder(std::string path, QueryData& results) {
 
     HRESULT lastTaskRun = E_FAIL;
     pRegisteredTask->get_LastTaskResult(&lastTaskRun);
-    _com_error err(lastTaskRun);
-    r["last_run_message"] = err.ErrorMessage();
+    LPWSTR messageBuffer = nullptr;
+    auto size = ::FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+                                     FORMAT_MESSAGE_FROM_SYSTEM |
+                                     FORMAT_MESSAGE_IGNORE_INSERTS,
+                                 nullptr,
+                                 lastTaskRun,
+                                 MAKELANGID(LANG_NEUTRAL, SUBLANG_NEUTRAL),
+                                 reinterpret_cast<LPWSTR>(&messageBuffer),
+                                 0,
+                                 nullptr);
+    std::wstring message(messageBuffer, size);
+    LocalFree(messageBuffer);
+    r["last_run_message"] = trim(wstringToString(message));
```

Comment:
As per #6304, recommend defining UNICODE and _UNICODE in build configurations. This way, `_com_error::ErrorMessage()` will properly return the correct text. You will have also fixed all other code that may be calling ANSI functions to return as TCHARs. If you look at `_com_error::ErrorMessage()` implementation in `comdef.h`, notice it is a bit more involved than a `FormatMessage` function call. It is more correct to not reimplement this method here: In the event that the underlying `comdef.h` implementation changes, we will not have received the benefit of that underlying change as per this code.